admin-manual/customization/authentication.rst

```diff
@@ -381,6 +381,391 @@ see:
+from both AtoM and the IAM server.
+
+Enabling OIDC authentication requires manually editing a few config files.
+For more information on how to do this, see :ref:`customization-config-files`
+
+.. NOTE::
+
+   Proper operation of the OIDC Plugin requires configuring AtoM with an external
+   cache. See :ref:`oidc-cache-configuration` for more information.
+
+Enable the plugin
+-----------------
+
+Create an empty file named ``activate-oidc-plugin`` in the root of the AtoM
+installation to enable the OIDC plugin. Permissions and ownership of this file
+should match the other folders in the AtoM folder.
+
+Add the plugin name to the ``$plugins`` array in ``config/ProjectConfiguration.class.php``
+
+.. image:: images/enable-oidc-plugin.*
+   :align: center
+   :width: 90%
+   :alt: Plugin name ``arOidcPlugin`` added in the ``$plugins`` array.
+
+.. IMPORTANT::
+   For security best practices, instead of having an administrator enabling the
+   plugin in the user interface, have a system administrator enable the plugin
+   as outlined above to avoid local authentication accounts stored in AtoM.
+
+Next, change the default login module to OIDC in ``apps/qubit/config/settings.yml``.
+
+.. image:: images/configure-oidc-login-module.*
+   :align: center
+   :width: 90%
+   :alt: Set ``login_module`` to ``oidc``.
+
+Finally, change the user class to ``oidcUser`` in ``apps/qubit/config/factories.yml``.
+
+.. image:: images/configure-oidc-user-class.*
+   :align: center
+   :width: 90%
+   :alt: Update user class to ``oidcUser``.
+
+Configure the OIDC settings in the plugin's ``app.yml`` file
+------------------------------------------------------------
+
+Next, configure the OIDC settings in the ``arOidcPlugin``'s ``app.yml`` config file
+with the details of the OIDC endpoint for authentication.
+
+.. WARNING::
+   While OIDC endpoint TLS validation may be disabled for development, it is
+   strongly discouraged to do so in a production environment.
+
+Update the ``providers`` array:
+
+.. image:: images/configure-oidc-settings-providers.*
+   :align: center
+   :width: 90%
+   :alt: Update the ``providers`` array in ``arOidcPlugin``'s ``app.yml``.
+
+One or many OIDC providers can be configured in the ``app.yml``. Each provider will
+have separate configuration for the following settings:
+
+.. NOTE::
+   If changing the primary provider name from the default ``primary``, uncomment
+   the setting ``primary_provider_name`` and specify the name of the primary provider.
+   Secondary providers can be named and triggered for use by specifying a query
+   param on the referring URL before pressing “Log in with SSO”. The query selector
+   name can be configured with the ``provider_query_param_name`` setting.
+
+**url**
+
+OIDC provider authentication redirect URL:
+
+Default for Dex in oidc-demo: http://dex:5556/dex
+
+Default for Keycloak direct in oidc-demo: http://keycloak:8080/realms/artefactual
```

Comment:
"This is the URL that you will be redirecting to. For example, using if redirecting to Keycloak this will look something like: 
`https://<keycloakURL>/realms/<atom_realm_name>"